### PR TITLE
Ticket651 add ioc cryosms

### PR DESCRIPTION
**Pull request by ChrisColeExpControl**
### Description of work
Added a simple test which checks that the IOC is behaving as intended. Checks that OUTPUTMODE changes when OUTPUTMODE:SP is changed. For this to happen, the setpoint must write to an asyn driver, which writes to a hidden PV, which uses streamdevice to write to an emulator, which is finally read by OUTPUTMODE through streamdevice.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/4826

### Acceptance criteria

The new test functions as described.

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### CODE REVIEW

File: tests/cryosms.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import unittest
+
+from utils.test_modes import TestModes
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.testing import get_running_lewis_and_ioc
+
+DEVICE_PREFIX = "CRYOSMS_01"
+EMULATOR_NAME = "HFMAGPSU"
```
Comment (DominicOram) [inline tests/cryosms.py]: I think this needs to be `cryogenic_sms`

File: tests/cryosms.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class CryoSMSTests(unittest.TestCase):
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc(EMULATOR_NAME, DEVICE_PREFIX)
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX, default_timeout=10)
+
+        self.ca.assert_that_pv_exists("DISABLE", timeout=30)
+
+    def test_WHEN_ioc_is_started_THEN_it_is_not_disabled(self):
+        self.ca.assert_that_pv_is("DISABLE", "COMMS ENABLED")
```
Comment (DominicOram) [inline tests/cryosms.py]: Why do we need this test?

Comment (ChrisColeExpControl) [reply tests/cryosms.py]: This was from the example file. Removed it here, though will be making a similar check in the next CRYOSMS ticket.

File: tests/cryosms.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class CryoSMSTests(unittest.TestCase):
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc(EMULATOR_NAME, DEVICE_PREFIX)
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX, default_timeout=10)
+
+        self.ca.assert_that_pv_exists("DISABLE", timeout=30)
+
+    def test_GIVEN_outputmode_sp_correct_WHEN_outputmode_sp_written_to_THEN_outputmode_changes(self):
```
Comment (Tom-Willemsen) [inline tests/cryosms.py]: This test is failing for me with the following error trace:

```
Running tests...
----------------------------------------------------------------------
F
======================================================================
FAIL [11.104s]: test_GIVEN_outputmode_sp_correct_WHEN_outputmode_sp_written_to_THEN_outputmode_changes (tests.cryosms.CryoSMSTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Instrument\Apps\EPICS\support\IocTestFramework\master\tests\cryosms.py", line 40, in test_GIVEN_outputmode_sp_correct_WHEN_outputmode_sp_written_to_THEN_outputmode_changes
    self.ca.assert_that_pv_is("OUTPUTMODE", new_outputmode, timeout=10)
  File "C:\Instrument\Apps\EPICS\support\IocTestFramework\master\utils\channel_access.py", line 259, in assert_that_pv_is
    pv, lambda val: val == expected_value, timeout=timeout, message=msg, pv_value_source=pv_value_source)
  File "C:\Instrument\Apps\EPICS\support\IocTestFramework\master\utils\channel_access.py", line 237, in assert_that_pv_value_causes_func_to_return_true
    raise AssertionError(err)
AssertionError: Expected PV, 'TE:NDW1799:CRYOSMS_01:OUTPUTMODE' to have value 'TESLA' (type: 'str').
Final PV value was 'AMPS' (type: 'str')

----------------------------------------------------------------------
Ran 1 test in 11.104s

FAILED (failures=1)
```

Note: I get the same error trace when testing IOC 2. When you have fixed this, please also check that IOC 2 is fixed.

**Pull request merged**